### DISCUSSION TITLE
Kidash program question

### DISCUSSION THREAD

**Discussion by ingmarthompson1964** [chaoss/grimoirelab#668] [General]
Hello Alvaro,
My name is Ingmar Thompson and I work for Dish Network in Cheyenne, WY.  I came across your kidash program on github and it looked promising as a solution to backup dashboards from our production Elasticsearch instances.  I have been playing around with your program and as you can see in the included images, the --list switch is working fine for me.  But when I try to export a dashboard by the dashboard_id, in debug the program returns that the dashboard cannot be found.  We are at version 8.13.2 which I am using to test your program.  Would you possibly have any insights into why your program is not finding the dashboard when trying to export it.  Please see the attached image files running on Rocky Linux for details of the command switches I am using for the --list and the --export commands?

Thank you for your time,
-Ingmar Thompson
Dish Network

En Espanol:
Mi nombre es Ingmar Thompson y trabajo para Dish Network en Cheyenne, WY. Descubrí tu programa kidash en GitHub y parecía prometedor como solución para hacer copias de seguridad de los paneles de control de nuestras instancias de Elasticsearch en producción. He estado probando tu programa y, como puedes ver en las imágenes adjuntas, el interruptor --list funciona bien para mí. Sin embargo, cuando intento exportar un panel de control por su ID de panel, en modo de depuración el programa devuelve que no se puede encontrar el panel de control. Estamos en la versión 8.13.2, que estoy usando para probar tu programa. ¿Tendrías alguna idea de por qué tu programa no encuentra el panel de control al intentar exportarlo? Por favor, consulta los archivos de imagen adjuntos ejecutándose en Rocky Linux para obtener detalles de los interruptores de comando que estoy usando para los comandos --list y --export. 
Gracias por tu tiempo, 
![kidash_cannot_find_specifc_dahboard_to_download](https://github.com/chaoss/grimoirelab/assets/96149071/b9607937-5911-4b6c-9595-6f08b94cbe23)
![kibana_list_specific_dashboard](https://github.com/chaoss/grimoirelab/assets/96149071/0a42717e-7535-4f25-a63c-6e26ce768144)
![kidash_list](https://github.com/chaoss/grimoirelab/assets/96149071/b35f0ebb-6f4f-4d1a-80f0-0a5c0c0a42c4)
-Ingmar Thompson
Dish Network

**Comment by ingmarthompson1964** [comment]
FYI... The password in image files has been changed...

**Comment by ingmarthompson1964** [comment]
OK, with adding further debugging it is looking like the function:
find_item_json(elastic, "dashboard", dashboard_id)
is using an API call which is no longer valid in version 8.13.2.

2024-04-30 14:46:21,166 The JSON item returned is {'error': 'no handler found for uri [/.kibana*/doc/dashboard:4c6e1ec0-38fd-11ed-94ef-1bde9482e397] and method [GET]'}
2024-04-30 14:46:21,166 Can not find type dashboard item dashboard:4c6e1ec0-38fd-11ed-94ef-1bde9482e397

I think the modern api call to get individual dashboards in now in saved_objects:

https://www.elastic.co/guide/en/kibana/8.13/saved-objects-api-export.html

**Comment by sduenas** [comment]
Kidash doesn't work for ElasticSearch >= 7. In fact, GrimoireLab won't support ElasticSearch in the near future. You can always contribute with some changes to Kidash to support other version but I'm afraid to say that the development of the tool has stopped.
